scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -225,6 +237,27 @@ public DTLSConnector(final DtlsConnectorConfig configuration, final SessionCache
 	}
 
 	/**
+	 * Sets the executor to use for processing records.
+	 * <p>
+	 * If this property is not set before invoking the {@linkplain #start() start method},
+	 * a new {@link StripedExecutorService} is used with a thread pool of size #(CPU cores).
+	 * <p>
+	 * In a production environment the thread pool used by the executor should be much larger
+	 * than just the number of CPU cores. This helps with performing multiple handshakes
+	 * in parallel, in particular if the key exchange requires look up of identities, e.g.
+	 * in a database or using a web service.
+	 * <p>
+	 * If this method is used to set an executor, the executor will <em>not</em> be shut down
+	 * by the {@linkplain #stop() stop method}.
+	 * 
+	 * @param executor The executor.
+	 */
+	public void setExecutor(StripedExecutorService executor) {
```

Comment:
Should we prevent setting the executor while the connector is running? Otherwise we get concurrency issues with hasInternalExecutor flag. 

Is it even necessary to have this setter instead of setting it in the constructor? Then we could make the executor field even final. atm we are accessing the executor from different threads without appropriate locking or making it volatile.